neo4j/internal/bolt/hydrator.go

```diff
@@ -1072,6 +1077,123 @@ func parseGqlStatusObject(m map[string]any) db.GqlStatusObject {
 	return g
 }
 
+func (h *hydrator) vector(n uint32) any {
+	h.assertLength("vector", 2, n)
+	if h.getErr() != nil {
+		return nil
+	}
+
+	// Read type marker
+	h.unp.Next()
+	if h.unp.Curr != packstream.PackedByteArray {
+		h.setErr(&db.ProtocolError{
+			MessageType: "vector",
+			Err:         "Expected byte array for vector type marker",
+		})
+		return nil
+	}
+	typeMarker := h.unp.ByteArray()
+	if len(typeMarker) != 1 {
+		h.setErr(&db.ProtocolError{
+			MessageType: "vector",
+			Err:         "Vector type marker must be exactly 1 byte",
+		})
+		return nil
+	}
+
+	// Read values
+	h.unp.Next()
+	if h.unp.Curr != packstream.PackedByteArray {
+		h.setErr(&db.ProtocolError{
+			MessageType: "vector",
+			Err:         "Expected byte array for vector values",
+		})
+		return nil
+	}
+	values := h.unp.ByteArray()
+
+	// Determine vector type and create appropriate vector
+	switch typeMarker[0] {
+	case 0xc1: // FLOAT_64
+		if len(values)%8 != 0 {
+			h.setErr(&db.ProtocolError{
+				MessageType: "vector",
+				Err:         "Vector values must be multiple of 8 bytes for float64",
+			})
+			return nil
+		}
+		result := make(dbtype.Vector[float64], len(values)/8)
+		for i := range result {
+			result[i] = math.Float64frombits(binary.BigEndian.Uint64(values[i*8 : (i+1)*8]))
+		}
```

Comment:
Pending actual benchmarking, I suspect it's faster to just allocate, but not zero the memory. Something along the lines of

```suggestion
		result := make(dbtype.Vector[float64], 0, len(values)/8)
		for i := range len(values)/8 {
			result = append(result, math.Float64frombits(binary.BigEndian.Uint64(values[i*8 : (i+1)*8])))
		}
```

Similar for the other types, won't repeat the comment.

Reply:
Benchmarking on my machine resulted in a speed-up in the order of 5-10%.

Reply:
Nice one, I've updated similar code too.